refactor: remove unused session_id parameter from API calls

Noor-Serry opened the pull request:
there are was exception thrown when try to get the session id and the user select continue as guest so that I removed this exception and and make it return default value if it was null

This change simplifies the code and eliminates unnecessary data transmission.

# Pull Request Template

## Description
Provide a clear and concise description of what this pull request does. Include the purpose and context for the changes.

---

## Changes Made
List the changes introduced in this pull request:

- AuthenticationLocalDataSourceImpl

---

## Screenshots (if applicable)
Include screenshots or GIFs to showcase the changes, especially if they impact the UI.
Before :

https://github.com/user-attachments/assets/a2cb0f73-b3e4-4096-b1ac-f4f3415fd58d

After:

https://github.com/user-attachments/assets/92b881a7-7029-4ba9-946e-ca9e195c2559

---

## Checklist
Please ensure the following tasks are completed:

- [x] My code follows the code style of this project
- [x] Changes have been tested manually and verified.
- [x] PR includes at most one single feature.

---

## Additional Comments
Add any additional information or context about the pull request here.

Noor-Serry committed:
refactor: remove unused session_id parameter from API calls

The `session_id` parameter was being passed to several API endpoints for fetching movie and TV show details (`getMovieDetailsById`, `getTvShowDetailsById`) but was not actually being used by the backend. This commit removes the `session_id` parameter from these API calls and their corresponding function definitions in the remote data sources, repositories, and API service interfaces.

This change simplifies the code and eliminates unnecessary data transmission.

Belal-Alkhatib commented on remoteDatasource/src/main/java/com/amsterdam/remotedatasource/datasource/MovieRemoteDataSourceImpl.kt:
```diff
@@ -38,8 +38,8 @@ class MovieRemoteDataSourceImpl @Inject constructor(
 
-    override suspend fun getMovieDetailsById(movieId: Long, sessionId: String): RemoteMovieDetailsResponse {
-        return responseCall { movieApiService.getMovieDetailsById(movieId, sessionId) }
+    override suspend fun getMovieDetailsById(movieId: Long): RemoteMovieDetailsResponse {
+        return responseCall { movieApiService.getMovieDetailsById(movieId) }
```
**Why did you remove this? It is necessary for retrieving the users rating.**
the `session_id `is already optional, so if your feature does not require it, you can simply ignore it. however, removing this breaks the ability to fetch the users personal rating when a session exists.

Belal-Alkhatib replied:
@Noor-Serry  please check if the rate dialog correctly displays the previous rating when you open the details screen of a movie you’ve already rated?

Noor-Serry replied:
great Belal I will revert it . I was wondering what is the purpose of sending it but now I know it purpose so that I will revert it and fix this bug with other way

saraashraf11 commented on remoteDatasource/src/main/java/com/amsterdam/remotedatasource/api/MovieApiService.kt:
```diff
@@ -52,7 +52,6 @@ interface MovieApiService {
     @GET(MOVIE_DETAILS_ENDPOINT)
     suspend fun getMovieDetailsById(
         @Path("movieId") movieId: Long,
-        @Query("session_id") sessionId: String = "",
```
as Bilal said, make sure that rate dialog correctly displays the previous rating

mohamedd-hassan reviewed (changes requested):
<img width="741" height="597" alt="image" src="https://github.com/user-attachments/assets/a461ada7-f05f-420e-96c6-235e9c25a23a" />

we had a session id thats defaulted to null in the retrofit client, consider fetching it from the preferences like the language and appending it to each call and the api will just ignore it if it does not need it

Noor-Serry commented:
@mohamedd-hassan the session id we dose not need it in all Api request unlike the language

mohamedd-hassan commented:
> @mohamedd-hassan the session id we dose not need it in all Api request unlike the language

@Noor-Serry not every api call needs the language parameter (the authentication for example) but it makes it so that not every repository fetches the language and appends it

here its the same case instead of every repository needing to remember to get the session_id from the preferences and decrypt it to send it, you just append it and assume all your calls are authenticated

Noor-Serry committed:
refactor: Add sessionId to movie/TV show details API calls

This commit modifies the `getMovieDetailsById` and `getTvShowDetailsById`
functions in the remote data sources and their corresponding API service
interfaces to include an optional `sessionId` parameter.

This change is necessary to fetch user-specific data, such as ratings,
when retrieving details for movies and TV shows.

The `AuthenticationLocalDataSourceImpl` has also been updated to return
an empty string instead of throwing an `UnauthorizedException` when the
session ID is not found, allowing for unauthenticated details fetching.

Additionally, the `MovieRepositoryImpl` and `TvShowRepositoryImpl` have been
updated to cache movie and TV show categories respectively if they are not
already cached locally. This ensures that category information is available
when saving movies/TV shows with their associated categories.

Noor-Serry committed:
Merge branch 'develop' of github.com:Amsterdam-Team/Aflami into fix/movie-tvshow-details-do-not-show-data-as-guest

# Conflicts:
#	repository/src/main/java/com/amsterdam/repository/repository/MovieRepositoryImpl.kt
#	repository/src/main/java/com/amsterdam/repository/repository/TvShowRepositoryImpl.kt

Noor-Serry committed:
Merge branch 'develop' of github.com:Amsterdam-Team/Aflami into fix/movie-tvshow-details-do-not-show-data-as-guest

# Conflicts:
#	repository/src/main/java/com/amsterdam/repository/repository/MovieRepositoryImpl.kt
#	repository/src/main/java/com/amsterdam/repository/repository/TvShowRepositoryImpl.kt

Noor-Serry committed:
Merge remote-tracking branch 'origin/fix/movie-tvshow-details-do-not-show-data-as-guest' into fix/movie-tvshow-details-do-not-show-data-as-guest

Noor-Serry commented:
@mohamedd-hassan the session id is some thing sensitive unlike the language

mohamedd-hassan commented:
@Noor-Serry its not like we send it in a different way in the other methods, also its not as sensitive as you might think. this session id even if intercepted would not work unless its used with the api key that generated it (if i took a session id you made with my api key it will give me unauthorized error) plus we are sending it all over https

actually it being sensitive is more reason for the client to handle it

i have an idea if you want, you can let the client handle it but have a custom header in the request ("use session") and in the client check if the request has that header it can append it so its only appended in the requests that need it without each repository having to fetch it from the datastore and decrypt it
